ocaml/typing/typecore.ml

```diff
@@ -3339,9 +3347,16 @@ let collect_unknown_apply_args env funct ty_fun mode_fun rev_args sargs ret_tvar
                               (expand_head env funct.exp_type)))
         in
         let arg =
-          Unknown_arg { sarg; ty_arg_mono; mode_fun; mode_arg; sort_arg }
+          match sarg.pexp_desc with
+          | Pexp_extension({txt = "extension.dummy"; loc}, PStr []) ->
+              if not (Language_extension.is_enabled Dummy_arguments) then
+                raise (Typetexp.Error (loc, env,
+                  Unsupported_extension Dummy_arguments));
+              Dummy {ty_arg; mode_fun; mode_arg; sort_arg; level = lv}
```

Comment:
I would prefer for the language extension check to be moved into the Jane Syntax module along with the literal extension string, but I don't feel too strgonly about this.